### PR TITLE
Add v1.7.0 template upgrade guide

### PR DESCRIPTION
**Pull request by sylviamoss**
Check out the preview: https://packer-o6yr91z21.vercel.app/guides/1.7-template-upgrade

### CODE REVIEW

File: website/content/guides/1.7-template-upgrade.mdx [hunk 1]
```diff
@@ -0,0 +1,154 @@
+---
+page_title: Upgrading your template to use Packer init
+sidebar_title: Upgrade Your Template to use Packer init
+---
+
+# Upgrading your template to be compatible with `packer init`
+
+The `packer init` command is introduced in version v1.7.0 at the same time the [Packer Plugin SDK](https://github.com/hashicorp/packer-plugin-sdk)
+is officially released. The Packer Plugin SDK is a change that benefits plugins maintainers, whilst the `init` command
+benefits Packer users by providing a better control of plugin usage within a Packer template configuration.
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:10]:
```suggestion
The `packer init` command introduced in version v1.7.0 benefits Packer users by providing better control of plugin usage within a Packer template configuration.
```
I think we can drop the call out for the SDK here as it release time is not relevant for this guide. The call out you have below `"The easiest way to know if the plugin you use fits those requirements, is by looking for SDK migration in their CHANGELOG"` is perfect and helps make this point.

File: website/content/guides/1.7-template-upgrade.mdx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+In short words, `packer init` will look at the `required_plugins` block definition within a template, take the required
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:12]:
```suggestion
In short words, `packer init` will look at the `required_plugins` block definition within a template, for the required
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+plugins specifications and download the correct binaries for that template. For details about how the command works and
+where the plugins will be installed, please refer to the [`packer init`](/docs/commands/init) documentation.
+
+-> **Note:** `packer init` does not work with legacy JSON templates. You can
+upgrade your JSON config files to HCL using the [hcl2_upgrade](/docs/commands/hcl2_upgrade) command.
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:17]:
```suggestion
-> **Note:** `packer init` is only supported for HCL templates. You can
upgrade legacy JSON templates to HCL using the [hcl2_upgrade](/docs/commands/hcl2_upgrade) command.
```

Nit: I find positives statements easier to process.

File: website/content/guides/1.7-template-upgrade.mdx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+## Updating your template with the `required_plugins` block
+
+### Why do I need to upgrade my template?
+
+To use the new released `packer init` command and have all the winnings that it is supposed to bring, you must upgrade
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:23]:
```suggestion
To use the `packer init` command and have all the winnings that it brings, you must upgrade
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+your template to use the `required_plugins` block. The `init` command won't work if no `required_plugins` is provided.
+
+We strongly encourage you to upgrade and start using `required_plugins` template, but if you are not confident on doing
+this now you can still install plugins manually like the old times.
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:27]:
```suggestion
We strongly encourage you to upgrade and start using `required_plugins` block within your templates to manage plugin installation, but if you prefer not to use the `required_plugins` block you can continue to [install plugins manually](/docs/plugins#installing-plugins).
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+### When to upgrade my template?
+
+The `packer init` can only install plugins that are upgraded to use our brand new [Plugin SDK](https://github.com/hashicorp/packer-plugin-sdk),
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:31]:
```suggestion
The `packer init` command can only install plugins that have upgraded to use the latest version of the [Packer Plugin SDK](https://github.com/hashicorp/packer-plugin-sdk),
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+and therefore compatible with our API version v5.0. The easiest way to know if the plugin you use fits those
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:32]:
```suggestion
and therefore compatible with Packer's API version v5.0. The easiest way to know if the plugin you use fits those
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+requirements, is by looking for SDK migration in their CHANGELOG, and noticing if the plugin repository is named as
+`packer-plugin-*` (e.g. `packer-plugin-comment`).
+If the plugin you use is upgraded, you are ready to upgrade your template.
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:35]:
```suggestion
If the plugin(s) have been upgraded, the can be used with the `required_plugins` block in your upgraded template.
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+
+With the SDK released at the same time as the `init` command, plugin users may face a gap between the release and
+plugins upgrade. Maintainers will have to upgrade their plugins, and the process is a little more complicated than this one.
+The Packer team is getting in touch with all currently known plugin maintainers to provide support during their upgrade process.
+
+If you are willing to upgrade your template but found out that the plugin you are using hasn't been upgraded yet,
+we suggest you reach out to the plugin maintainers and ask for an upgrade.
+
+Check it out the table below to better understand what behavior to expect from a Packer version with the current state
+of your template, and the plugins you are using.
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:47]:
```suggestion
Check the table below to better understand what behavior to expect from a Packer version with the current state
of your template and the plugins you are using.
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+    ui = true
+    bubble_text =  true
+  }
+}
+```
+
+This template uses the `packer-provisioner-comment` plugin and, until Packer v1.7.0, the binaries were installed manually.
+
+From Packer v1.7.0, the plugin has been upgraded by its maintainers and now is called `packer-plugin-comment` and the
+latest version is v0.2.23. Knowing that, our example template can be upgraded.
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:79]:
```suggestion
From Packer v1.7.0, the plugin has been upgraded by its maintainers and is now called `packer-plugin-comment` with it latest version being v0.2.23. Knowing that, our example template can be upgraded to use the `required_plugins` block.
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+  provisioner "comment" {
+    comment = "Hello from comment plugin"
+    ui = true
+    bubble_text =  true
+  }
+}
+```
+
+The upgraded template is now telling Packer that it requires a plugin from the source `sylviamoss/comment`
+of the version equals or greater `v0.2.23`. It also tells that the plugin should be referred in the rest of the template as `comment`.
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:109]:
```suggestion
with the version equal to or greater than `v0.2.23`. It also indicates that the plugin should be referred as `comment` in the rest of the template.
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+
+Here it is a brief explanation of each field:
+- `version` - Should follow the [version constraints](/docs/templates/hcl_templates/blocks/packer#version-constraints).
+- `source` - Should have the plugin's organizational namespace on GitHub and its short name. Packer will be responsible
+to write the full GitHub address.
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:114]:
```suggestion
for determining the full GitHub address.
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+
+Once the template is upgraded, you can run `packer init example.pkr.hcl` and the output should tell you the installed
+plugin, and the place where it was installed.
+
+```shell
+➜ packerdev init example.pkr.hcl
+Installed plugin sylviamoss/comment v0.2.23 in "/Users/<user>/.packer.d/plugins/github.com/sylviamoss/comment/packer-plugin-comment_v0.2.23_x5.0_darwin_amd64"
+```
+
+`packer init` will only download plugins binary and never delete existing ones. If all the required_plugins are installed,
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:147]:
```suggestion
`packer init` will only download plugin binaries and never delete existing ones. If all the required_plugins are installed,
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+the command won't do anything unless you set the `-upgrade` flag to force download newer versions of the existing plugins.
+
+Now that the required comment plugin is installed via `packer init`, you can run `packer build example.pkr.hcl` as usual
+and won't need to run `init` again unless you want to upgrade the plugin version.
```
Comment (nywilken) [inline website/content/guides/1.7-template-upgrade.mdx:151]: Should we add a little blurb about calling build before init?

**Review by nywilken — APPROVED** [review]
This looks great. I left a couple of suggestions and small fixes.

File: website/content/guides/1.7-template-upgrade.mdx (hunk 13 above)
Comment (sylviamoss) [reply website/content/guides/1.7-template-upgrade.mdx:147]: I never know how to write this which one should be plural 🤦🏼

File: website/content/guides/1.7-template-upgrade.mdx (hunk 1 above)
Comment (sylviamoss) [reply website/content/guides/1.7-template-upgrade.mdx:10]: Great! If you felt that was enough then you right. I wrote this part before right the other one.

File: website/content/guides/1.7-template-upgrade.mdx (hunk 3 above)
Comment (sylviamoss) [reply website/content/guides/1.7-template-upgrade.mdx:17]: Oh yeah, sounds much better!

File: website/content/guides/1.7-template-upgrade.mdx (hunk 14 above)
Comment (sylviamoss) [reply website/content/guides/1.7-template-upgrade.mdx:152]: Just added one at the end of the guide. Let me know if there's something to change/fix there.

Comment (nywilken) [reply website/content/guides/1.7-template-upgrade.mdx:152]: Love it! Sounds great.

**Review by nywilken — APPROVED** [review]
Looks great to me :smile:

File: website/content/guides/1.7-template-upgrade.mdx [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+page_title: Upgrading your template to use Packer init
+sidebar_title: Upgrade Your Template to use Packer init
+---
+
+# Upgrading your template to be compatible with `packer init`
+
+The `packer init` command introduced in version v1.7.0 benefits Packer users by providing better control of plugin
+usage within a Packer template configuration.
+
+In short words, `packer init` will look at the `required_plugins` block definition within a template, for the required
```
Comment (SwampDragons) [inline website/content/guides/1.7-template-upgrade.mdx:11]:
```suggestion
In short, `packer init` will look at the `required_plugins` block definition within a template, for the required
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 16]
```diff
@@ -0,0 +1,157 @@
...
+where the plugins will be installed, please refer to the [`packer init`](/docs/commands/init) documentation.
+
+-> **Note:** `packer init` is only supported for HCL templates. You can
+upgrade legacy JSON templates to HCL using the [hcl2_upgrade](/docs/commands/hcl2_upgrade) command.
+
+## Updating your template with the `required_plugins` block
+
+### Why do I need to upgrade my template?
+
+To use the `packer init` command and have all the winnings that it brings, you must upgrade
```
Comment (SwampDragons) [inline website/content/guides/1.7-template-upgrade.mdx:22]:
```suggestion
To use the `packer init` command, you must upgrade
```

File: website/content/guides/1.7-template-upgrade.mdx [hunk 17]
```diff
@@ -0,0 +1,157 @@
...
+your template to use the `required_plugins` block. The `init` command won't work if no `required_plugins` is provided.
+
+We strongly encourage you to upgrade and start using `required_plugins` block within your templates to manage plugin
```
Comment (SwampDragons) [inline website/content/guides/1.7-template-upgrade.mdx:25]:
```suggestion
We strongly encourage you to upgrade and start using the `required_plugins` block within your templates to manage plugin
```

**Review by SwampDragons — COMMENTED** [review]
This is really good, but missing what I think is our most common use case: users who only use the plugins currently bundled with the core. We should explicitly tell them what they need to do now and what they can expect to need to do in the future.

**Comment by SwampDragons** [comment]
ugh I clicked "add review" before I was done reading -- I just added that note to remind myself the check later in the doc whether it was true; still reading so if it turns out I'm wrong sorry for jumping the gun

**Review by SwampDragons — APPROVED** [review]
I have a lot of little grammatical nitpicks, but this is great -- rather than doing the nitpicks as individual inline suggestions, I'm gonna PR them.

**Pull request merged**